### PR TITLE
PR template added

### PR DESCRIPTION
**Pull request by sadath-12**
Added pr template as suggested in #449

### CODE REVIEW

**Review by daixiang0 — APPROVED** [review]

**Review by mikeee — CHANGES_REQUESTED** [review]
I believe the descriptions or context should be in a comment so can be ignored on open rather than having to remove/delete the placeholder text.

**Comment by sadath-12** [comment]
> Me

> I believe the descriptions or context should be in a comment so can be ignored on open rather than having to remove/delete the placeholder text.

Just felt if we need to make sure the contributor must always provide information of what he did instead of making it optional . Please let me know if I am thinking it in right way

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Description
+
+_Please explain the changes you've made_
```
Comment (daixiang0) [inline .github/pull_request_template.md:3]: Hide it.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+## Issue reference
+
+We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.
```
Comment (daixiang0) [inline .github/pull_request_template.md:7]: Same as above.

**Comment by sadath-12** [comment]
thanks done @daixiang0

**Comment by mikeee** [comment]
To clarify my opinion is that we should comment out the placeholder/descriptive text so that we provide an indication of what to include and does not include it.

**Comment by mikeee** [comment]
To clarify - my opinion is that we should comment out the placeholder/descriptive text so that we provide an indication of what to include and does not include it.

**Comment by daixiang0** [comment]
> To clarify - my opinion is that we should comment out the placeholder/descriptive text so that we provide an indication of what to include and does not include it.

Agree, @sadath-12 please comment rather than delete.

**Comment by sadath-12** [comment]
yeah I get it now thanks @daixiang0  @mikeee

**Review by daixiang0 — APPROVED** [review]

**Pull request merged**